modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/dbf/DbaseFileReader.java

```diff
@@ -181,6 +181,23 @@ private void init(
             final TimeZone timeZone)
             throws IOException {
         this.channel = dbfChannel;
+        boolean initialized = false;
```

Comment:
I see you're trying to clean up in case the initialization fails. Per se this is fine.
However the approach is bothering me, what I see:
* Keep one field initialization in the original method (arbitrary)
* Create a new overloaded method with the same name and slightly different signature, which is actually doing the risky job and initializing some other fields

Could you make things clearer by:
* Creating a new method, calling it "initSafe" (or other name of your choice), do only the try/finally bits in there
* Call the the old init, un-modified

(same goes for the other changes following the same pattern).